lib/presentation/mixins/contacts_mixin.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:debounce_throttle/debounce_throttle.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/contact_manager/contact_manager.dart';
+import 'package:fluffychat/domain/model/extensions/contact/contacts_extension.dart';
+import 'package:fluffychat/presentation/extensions/contact/presentation_contact_extension.dart';
+import 'package:fluffychat/presentation/model/presentation_contact.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+
+mixin class ContactsMixin {
+  static const _debouncerIntervalInMilliseconds = 300;
+
+  final TextEditingController textEditingController = TextEditingController();
+
+  // FIXME: Consider can use FocusNode instead ?
+  final ValueNotifier<bool> isSearchModeNotifier = ValueNotifier(false);
+
+  final ValueNotifier<List<PresentationContact>> presentationContactNotifier =
+      ValueNotifier([]);
+
+  final FocusNode searchFocusNode = FocusNode();
+
+  final Debouncer _debouncer = Debouncer(
+    const Duration(milliseconds: _debouncerIntervalInMilliseconds),
+    initialValue: '',
+  );
+
+  final contactManager = getIt.get<ContactManager>();
+
+  void initialFetchContacts() async {
+    presentationContactNotifier.addListener(() {
+      Logs().d(
+        'ContactManagerMixin()::initSearchContacts: ${presentationContactNotifier.value.length}',
+      );
+    });
+    textEditingController.addListener(() {
+      _debouncer.value = textEditingController.text;
+    });
+
+    _debouncer.values.listen((keyword) {
+      _handleSearchContacts(keyword);
+    });
+    if (contactManager.tomContacts.isEmpty) {
```

Comment:
no need check empty

Reply:
I'll update it:
1. Check `firstSynchronized` if == true => call sync else get contacts from `contactManager `